packages/astro/src/core/build/static-build.ts

```diff
@@ -172,8 +175,15 @@ async function ssrBuild(
 					assetFileNames: `${settings.config.build.assets}/[name].[hash][extname]`,
 					...viteConfig.build?.rollupOptions?.output,
 					entryFileNames(chunkInfo) {
-						if (chunkInfo.facadeModuleId === resolvedPagesVirtualModuleId) {
-							return opts.buildConfig.serverEntry;
```

Comment:
This is a configurable value by the user, so we need to preserve it some how for SSR usage. If the user isn't using serverless mode I think we can still use this as the entry point probably, right?

Reply:
This code path wasn't used for SSR but for SSG. In fact, `resolvedPagesVirtualModuleId` is the virtual module name assigned by `plugin-pages`. 

`serverEntry` is never used for SSR, so I belive we have a bug on hand (I haven't seen a test case with `serverEntry` anywhere).

Reply:
Oh right, disregard then.